image-metadata-addition/main.py

```diff
@@ -18,86 +19,133 @@ def start_kafka() -> None:
+    """
+    Builds an annotation out of the assertions created by the Pillow imaging library
+    :param image_assertions: assertions declared by library
+    :param digital_media: json of the digital-media object
+    :param timestamp: formatted date time
+    :return: List of annotations
+    """
+    annotations = list()
+    ods_agent = shared.get_agent()
+    oa_selector = shared.build_class_selector("$ods:hasAssertion")
 
+    for assertion in image_assertions:
+        annotation = shared.map_to_annotation(ods_agent, timestamp, assertion, oa_selector, digital_media[shared.ODS_ID],
+                                       digital_media[shared.ODS_TYPE], "https://pypi.org/project/pillow/")
+        annotations.append(annotation)
+    return annotations
 
-def send_updated_opends(annotations: list, producer: KafkaProducer) -> None:
-    for annotation in annotations:
-        logging.info('Publishing annotation: ' + str(annotation))
-        producer.send('annotation', annotation)
 
+def publish_annotation_event(annotation_event: Dict,
+                             producer: KafkaProducer) -> None:
+    """
+      Send the annotation to the Kafka topic
+      :param annotations: The formatted list of annotations
+      :param producer: The initiated Kafka producer
+      :return: Will not return anything
+      """
+    logging.info('Publishing annotation: ' + str(annotation_event))
+    producer.send(os.environ.get('KAFKA_PRODUCER_TOPIC'), annotation_event)
 
-def get_image_info(image_uri: str) -> dict:
+
+def get_image_assertions(image_uri: str, timestamp: str) -> List[Dict]:
     """
     Checks if the Image url works and gathers metadata information from the image
     :param image_uri: The image url from which we will gather metadata
     :return: Returns a list of additional info about the image
     """
+    ods_agent = shared.get_agent()
+    assertions = list()
+    assertions.append(
+        build_assertion(timestamp, ods_agent, 'ac:variant', 'acvariant:v008',
+                        None))
     try:
         img = Image.open(requests.get(image_uri, stream=True).raw)
         img_file = BytesIO()
         img.save(img_file, img.format, quality='keep')
-        additional_info = {'exif:PixelXDimension': img.width,
-                           'exif:PixelYDimension': img.height,
-                           'dcterms:format': img.format.lower(),
-                           'dcterms:extent': str(round(img_file.tell() / 1000000, 2)) + 'MB',
-                           'ac:variant': 'acvariant:v008'}
+        assertions.append(
+            build_assertion(timestamp, ods_agent, 'exif:PixelXDimension', str(img.width),
+                            'pixel'))
+        assertions.append(
+            build_assertion(timestamp, ods_agent, 'exif:PixelYDimension', str(img.height),
+                            'pixel'))
+        assertions.append(
```

Comment:
I think this is actually a top level field in digital-media (`dcterms:format`), so then the annotation should be add or modifying.

Reply:
Not sure what you mean here. This assertion gets passed as the `oa_value` field in `shared.map_to_annotation()` ([link](https://github.com/DiSSCo/demo-enrichment-service-image/blob/ae0d4e227837c997049df88a2719de1f454bfe53/shared/shared.py#L75)), the motivation is `ods:adding` in that function. 

Do you mean these annotations should all have the field-selector target `dcterms:format`?

Reply:
Often modifying probably, as I hope it is present in most cases, otherwise adding

Reply:
Oh sorry, there is a top level field called `dcterms:format`, so instead of an assertion this can be a modifying/adding annotation on the top level field. Assertations is more for fields which did not get there own openDS field

Reply:
So we'd have 5 annotations on dcterms:format? How could that be integrated into the specimen? dcterms:format is one string, not an array, so i'm not sure how to apply all this additional information to one field. 

[dcterms:format](https://www.dublincore.org/specifications/dublin-core/dcmi-terms/#format):
> The file format, physical medium, or dimensions of the resource....Recommended practice is to use a controlled vocabulary where available. For example, for file formats one could use the list of Internet Media Types [[MIME](https://www.iana.org/assignments/media-types/media-types.xhtml)]. Examples of dimensions include size and duration.

Not sure what they mean by "examples of dimensions" here, couldn't find a MIME type.

Reply:
I could do 5-6 simple annotations with oa:values like: 
```
{
    "exif:PixelXDimension": img.width
}
```
for each measured property, and use a fieldSelector where field=dcterms:format, is that what you mean?

Reply:
but then that information wouldn't be able to be applied to the specimen, it would stay as an annotation. I was thinking assertions would be better because as assertions, the annotation could be integrated into the specimen